### PR TITLE
Java: Add `FCALL_RO` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
`Ctrl+C` `Ctrl+V` from https://github.com/aws/glide-for-redis/pull/1543

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsBaseCommands.java [hunk 1]
```diff
@@ -13,12 +14,17 @@
 public interface ScriptingAndFunctionsBaseCommands {
 
     /**
-     * Invokes a previously loaded function.
+     * Invokes a previously loaded function.<br>
+     * This command is routed to primary nodes only.<br>
+     * Always routed to a primary node, to route to a replica please refer to {@link #fcallReadOnly}.
+     * <br>
+     * TODO https://github.com/amazon-contributing/redis-rs/pull/150#issuecomment-2158875720 and
+     * https://github.com/aws/glide-for-redis/pull/1528/files#diff-3d2fda4dc5a83776d549c973097be5039ca22a9afc729c2ca460924a8e65d0f5R381
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsBaseCommands.java:22]: TBD

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsBaseCommands.java:22]: Bar's reply: 
> Yes you can add to the documentation something like 
```
Always routed to a primary node, to route to a replica please refer to fcall_ro
```

- [x] update documentation

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsBaseCommands.java:22]: Please remove TODO.

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java [hunk 2]
```diff
@@ -266,6 +267,7 @@ CompletableFuture<ClusterValue<Map<String, Object>[]>> functionList(
      */
     CompletableFuture<String> functionDelete(String libName, Route route);
 
+    // TODO update FCALL docs
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java:270]: Please remove TODO.

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsCommands.java [hunk 3]
```diff
@@ -129,7 +130,12 @@ public interface ScriptingAndFunctionsCommands {
     CompletableFuture<String> functionDelete(String libName);
 
     /**
-     * Invokes a previously loaded function.
+     * Invokes a previously loaded function.<br>
+     * This command is routed to primary nodes only.<br>
+     * Always routed to a primary node, to route to a replica please refer to {@link #fcallReadOnly}.
+     * <br>
+     * TODO https://github.com/amazon-contributing/redis-rs/pull/150#issuecomment-2158875720 and
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsCommands.java:137]: Please remove TODO.

File: java/integTest/src/test/java/glide/cluster/CommandTests.java [hunk 4]
```diff
@@ -842,6 +846,14 @@ public void function_commands_without_keys_with_route(boolean singleNodeRoute) {
                 assertEquals("one", nodeResponse);
             }
         }
+        fcallResult = clusterClient.fcallReadOnly(funcName, new String[] {"one", "two"}, route).get();
+        if (route instanceof SingleNodeRoute) {
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/cluster/CommandTests.java:850]: you could also use `fcallResult.hasMultiData`...

File: java/integTest/src/test/java/glide/cluster/CommandTests.java [hunk 5]
```diff
@@ -947,6 +959,15 @@ public void function_commands_without_keys_with_route(boolean singleNodeRoute) {
                 assertEquals(2L, nodeResponse);
             }
         }
+        fcallResult =
+                clusterClient.fcallReadOnly(newFuncName, new String[] {"one", "two"}, route).get();
+        if (route instanceof SingleNodeRoute) {
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/cluster/CommandTests.java:964]: you could also use `fcallResult.hasMultiData`...

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**